File: Bachelor/Semester4/Systems_for_Design_and_Implementation/C#/ProiectCShMPP/Server/service/UserService.cs
using log4net;
using ProiectCShMPP.domain;
using ProiectCShMPP.repository;

namespace ProiectCShMPP.service
{
    public class UserService
    {
        private readonly IUserRepository _repository;
        private static readonly ILog Logger = LogManager.GetLogger(typeof(UserService));

        public UserService(IUserRepository repository)
        {
            _repository = repository;
        }

        /// <summary>
        /// Method for finding a user based on its username or password (e.g. login)
        /// </summary>
        /// <param name="username">the desired username</param>
        /// <param name="password">the desired password</param>
        /// <returns>
        /// - null, if there is no User with username and password
        /// - the user, otherwise
        /// </returns>
        public User Find(string username, string password)
        {
            Logger.InfoFormat("Entry find user {0} {1}", username, password);
            User result = _repository.FindByUsernamePassword(username, password);
            Logger.InfoFormat("Exit find user {0}", result);
            return result;
        }
        
    }
}

File: Bachelor/Semester4/Software_engineering/Project/Server/src/main/java/service/UserService.java
package service;

import domain.User;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import repository.UserRepository;

import java.util.Optional;

public class UserService {
    private final UserRepository repository;
    private final static Logger logger = LogManager.getLogger();

    public UserService(UserRepository repository) {
        this.repository = repository;
    }

    /**
     * Method for finding a user by their username and password
     * @param username: String, the desired username
     * @param password: String, the desired password
     * @return an {@code Optional}
     *          - null, if there is no user with that {@code username} and {@code password}
     *          - the user, otherwise
     */
    public Optional<User> find(String username, String password){
        logger.traceEntry("Entry Find {} {}", username, password);
        Optional<User> result = repository.findByUsernamePassword(username, password);
        logger.traceExit("Exit Find result {}", result);
        return result;
    }
}
